tikv/tikv
# log-backup: make PITR available when using partial cert chain

### What is changed and how it works?

Issue Number: Close #13959

What's Changed:
This PR sets [`X509_V_FLAG_PARTIAL_CHAIN`](https://man.openbsd.org/X509_VERIFY_PARAM_set_flags.3#VERIFICATION_FLAGS), so we would trust the CA even there isn't a root CA provided.
Given the CA file is provided by the user, I think it should be fully trusted. Also note that the `grpc-io` library sets this flag by default.


```commit-message
This PR sets X509_V_FLAG_PARTIAL_CHAIN, so we would trust the CA even there isn't a root CA provided.
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
We made a binary for testing this, this is the code snippet:
```rust
    let cfg = Config::parse();
    let security_mgr = Arc::new(
        SecurityManager::new(&cfg.clone().into()).expect("failed to create security manager."),
    );
    println!("using config = {:?}", cfg);
    let ccfg = ConnectionConfig {
        keep_alive_interval: Duration::from_secs(3),
        keep_alive_timeout: Duration::from_secs(10),
        tls: Some(
            security_mgr
                .client_suite()
                .expect("parse client suite failed"),
        ),
    };
    println!("using connection config = {:?}", ccfg);
    let etcd_cli = LazyEtcdClient::new(cfg.endpoints.as_slice(), ccfg);
    let client_id = rand::thread_rng().gen::<u64>();
    println!("using client_id = {}", client_id);
    let t = tokio::runtime::Builder::new_current_thread()
        .enable_all()
        .build()
        .unwrap();
    let kv = KeyValue(
        MetaKey("/hello/world".as_bytes().to_vec()),
        client_id.to_string().into_bytes(),
    );
    println!("setting {:?} to PD", kv);
    t.block_on(etcd_cli.set(kv)).unwrap();
    println!("success!");
```

### Release note 

```release-note
Fixed a bug that caused PITR cannot work in some special TLS environment.
```
